File: code/python/322.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        dp = [float('inf')] * (amount + 1)
        dp[0] = 0
        for coin in coins:
            for num in range(coin,amount+1):
                dp[num] = min(dp[num],dp[num-coin]+1)
        print(dp)
        return dp[-1]

File: code/java/322.java
class Solution {
    int max=Integer.MAX_VALUE;
     public int coinChange(int[] coins, int amount) {
         int[] dp = new int[amount+1];
         for(int i=1;i<=amount;i++) dp[i] = max;
         for(int coin:coins){
             for(int i=coin;i<=amount;i++){
                 dp[i] = Math.min(dp[i],dp[i-coin]+1);
             }
         }
         if(dp[amount]==max) return -1;
         else return dp[amount];
    }
}